### ISSUE TITLE
tf.nn.silu loses a finite second-derivative signal for float64 input

### ISSUE THREAD

**Issue by ALinrunrun** [tensorflow/tensorflow#126632]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

tf 2.21.0

### Custom code

Yes

### OS platform and distribution

Linux Ubuntu 22.04

### Python version

Python 3.13

### Current behavior?

`tf.nn.silu` returns zero for a second-derivative component whose correctly rounded `float64` reference value is finite, normal, and nonzero.

SiLU is `x * sigmoid(x)`, and its second derivative follows from differentiating this product. At `x = 40.410841513445845`, the high-precision reference rounded to float64 is approximately `-1.0820525268131133e-16`.

However, TensorFlow autodiff returns `0.0` for this second derivative. The forward value is finite and correctly represented; the discrepancy is in the second-order autodiff result.

Expected behavior?

The second derivative of `tf.nn.silu` at `x = 40.410841513445845` should be close to `-1.0820525268131133e-16`, not `0.0`.

### Standalone code to reproduce the issue

```shell
import os
os.environ.setdefault("CUDA_VISIBLE_DEVICES", "-1")
os.environ.setdefault("TF_CPP_MIN_LOG_LEVEL", "3")

import tensorflow as tf

x_value = 40.410841513445845
expected = -1.0820525268131133e-16
rtol = 1e-6

x = tf.Variable(x_value, dtype=tf.float64)

with tf.GradientTape() as outer:
    with tf.GradientTape() as inner:
        y = tf.nn.silu(x)
    g1 = inner.gradient(y, x)

g2 = outer.gradient(g1, x)

actual = float(g2)
rel_err = abs(actual - expected) / abs(expected)

print(f"x: {x_value!r}")
print(f"forward value: {float(y)!r}")
print(f"first derivative: {float(g1)!r}")
print(f"tensorflow second derivative: {actual!r}")
print(f"expected second derivative: {expected!r}")
print(f"relative error: {rel_err:.3e}")

if rel_err > rtol:
    print("BUG REPRODUCED: tf.nn.silu loses a finite second-derivative signal")
else:
    print("not reproduced")
```

### Relevant log output

```shell
x: 40.410841513445845
forward value: 40.410841513445845
tensorflow second derivative: 0.0
expected second derivative: -1.0820525268131133e-16
relative error: 1.000e+00
BUG REPRODUCED: tf.nn.silu loses a finite second-derivative signal
```

**Comment by Venkat6871** [comment]
Hi @ALinrunrun,

Thank you for reporting this issue and for providing a clear and reproducible example. You've demonstrated that `tf.nn.silu`'s second derivative returns `0.0` for a `float64` input where a small, finite value is expected, indicating a potential precision issue in higher-order autodiff.

We appreciate the detailed explanation and reproduction. We will investigate this issue from our side and provide an update.

**Comment by Venkat6871** [comment]
Hi @ALinrunrun  ,
Thank you for raising your concern here.
A fix for this issue has been proposed in this [PR](https://github.com/tensorflow/tensorflow/pull/126824). Once it is merged, this issue will be resolved.

Hi @GodlyDonuts ,
Thank you for raising the PR to address this issue. We truly appreciate your initiative and the effort in contributing the fix.
